Microsoft.Toolkit.Uwp.UI.Controls/TextBoxMask/TextBoxMask.cs

```diff
@@ -0,0 +1,303 @@
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void OnCustomMaskChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            InitTextBoxMask(d, e);
+        }
+
+        private static void InitTextBoxMask(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var textbox = d as TextBox;
+
+            // incase no value is provided us it as normal textbox
+            var mask = textbox?.GetValue(MaskProperty) as string;
+            if (string.IsNullOrWhiteSpace(mask))
+            {
+                return;
+            }
+
+            var placeHolderValue = textbox.GetValue(PlaceHolderProperty) as string;
+            if (string.IsNullOrEmpty(placeHolderValue))
+            {
+                return;
+            }
+
+            var placeHolder = placeHolderValue[0];
+
+            var representationDictionary = textbox.GetValue(RepresentationDictionaryProperty) as Dictionary<char, string>;
+            representationDictionary = new Dictionary<char, string>();
+            representationDictionary.Add(AlphaCharacterRepresentation.Key, AlphaCharacterRepresentation.Value);
+            representationDictionary.Add(NumericCharacterRepresentation.Key, NumericCharacterRepresentation.Value);
+            representationDictionary.Add(AlphaNumericRepresentation.Key, AlphaNumericRepresentation.Value);
+
+            var customDictionaryValue = textbox.GetValue(CustomMaskProperty) as string;
+            if (!string.IsNullOrWhiteSpace(customDictionaryValue))
+            {
+                var customRoles = customDictionaryValue.Split(',');
+                foreach (var role in customRoles)
+                {
+                    var roleValues = role.Split(':');
+                    if (roleValues.Length != 2)
+                    {
+                        throw new ArgumentException("Invalid CustomMask property");
+                    }
+
+                    var keyValue = roleValues[0];
+                    var value = roleValues[1];
+                    char key;
+
+                    // an exception should be throw if the regex is not valid
+                    Regex.Match(string.Empty, value);
+                    if (!char.TryParse(keyValue, out key))
+                    {
+                        throw new ArgumentException("Invalid CustomMask property, please validate the mask key");
+                    }
+
+                    representationDictionary.Add(key, value);
+                }
+            }
+
+            // TODO: insert generic custom representation
+            textbox.SetValue(RepresentationDictionaryProperty, representationDictionary);
+
+            var displayText = mask;
+            foreach (var key in representationDictionary.Keys)
+            {
+                displayText = displayText.Replace(key, placeHolder);
+            }
+
+            textbox.Text = displayText;
```

Comment:
Is displayText a hint to the user? This will set properties that are binding to the control. Probably best to have the user put their own watermark on.